client/setec/store.go

```diff
@@ -215,6 +225,37 @@ func (s *Store) secretLocked(name string) Secret {
 	return f
 }
 
+// LookupSecret returns a fetcher for the named secret. If name is already
+// known by s, this is equivalent to Secret; otherwise, s attempts to fetch the
+// latest active version of the secret from the service and either adds it to
+// the collection or reports an error.  LookupSecret does not automatically
+// retry in case of errors.
+func (s *Store) LookupSecret(name string) (Secret, error) {
+	if f := s.Secret(name); f != nil {
+		return f, nil
+	} else if !s.allowLookup {
+		return nil, errors.New("lookup is not enabled")
+	}
+
+	// Lock exclusive, as we need to modify the map with results.
+	s.active.Lock()
+	defer s.active.Unlock()
+	return s.lookupSecretLocked(name)
```

Comment:
If the secrets service is down or slow, doesn't this block all other access to secrets? Should this do the Get/etc. outside of the lock and only lock when it's ready to make structure updates?

Reply:
If it's properly down, then the connection should fail and we'll get an error. If it's just slow this _will_ block, but I figured the frequency with which we are likely to conflict is low enough that it shouldn't matter. But it would not be difficult to separate them, if you think it's worthwhile.